File: 900 Rating/firework.py
# not solved 

test = int(input())
while(test!=0):
    a,b,c=map(int,input().split())
    count1 = 0
    c+=1
    count2=0
    d = 1
    e = 1
    my_set = set()
    while(count1!=a):
        my_set.add(c+d*a)
        d+=1
        count1+=1
    while count2!=b:
        my_set.add(c+e*b)
        e+=1
        count2+=1

    print(len(my_set))


    test-=1

File: 900 Rating/firework.cpp
# include <bits/stdc++.h>
using namespace std;
// Solved 

int main()
{
    int test;
    cin>>test;
    for (int i=0;i<test; i++)
    {
        long long int a,b,c;
        cin>>a>>b>>c;
        long long int answer = 0;
        c+=1;
        if(c%a==0)
        {
            answer=answer+c/a;
        }
        else{
            answer=answer+c/a+1;
        }
        if(c%b==0)
        {
            answer=answer+c/b;
        }
        else{
            answer=answer+c/b+1;
        }
        cout<<answer<<'\n';
    }
}